test/unit/math/prim/mat/prob/categorical_logit_test.cpp

```diff
@@ -79,3 +81,58 @@ TEST(ProbDistributionsCategoricalLogit, error) {
   ns[1] = 12;
   EXPECT_THROW(categorical_logit_log(ns, theta), std::domain_error);
 }
+
+TEST(ProbDistributionsCategoricalLogit, error_check) {
+  using stan::math::categorical_logit_rng;
+  boost::random::mt19937 rng;
+
+  Matrix<double, Dynamic, 1> beta(3);
+
+  beta << 1.0, 10.0, -10.0;
+
+  EXPECT_NO_THROW(categorical_logit_rng(beta, rng));
+
+  beta << -1e3, 1.1e3, 1e5;
+
+  EXPECT_NO_THROW(categorical_logit_rng(beta, rng));
+
+  beta(1) = std::numeric_limits<double>::quiet_NaN();
+  EXPECT_THROW(categorical_logit_rng(beta, rng), std::domain_error);
+
+  beta(1) = std::numeric_limits<double>::infinity();
+  EXPECT_THROW(categorical_logit_rng(beta, rng), std::domain_error);
+}
+
+TEST(ProbDistributionsCategoricalLogit, chiSquareGoodnessFitTest) {
+  boost::random::mt19937 rng;
```

Comment:
I don't see what setting seeds would do.  We don't need to get the exact same answer every time here, just make sure we're not getting unexpected results.